src/main/java/com/example/restapp/dao/AbstractDAO.java

```diff
@@ -4,13 +4,13 @@
 
 public abstract class AbstractDAO {
 
-	protected BeanSession session;
-
-	public void setBeanSession(BeanSession session) {
+	private final BeanSession session;
```

Comment:
I vote to change to protected. `protected final` is fine. Non-final not so much. Please go ahead, commit and push to this branch. Thanks!